src/elements/content-sidebar/Sidebar.js

```diff
@@ -112,11 +126,13 @@ class Sidebar extends React.Component<Props, State> {
         const { file_version: currentVersion } = file;
         const fileVersionSlug = currentVersion ? `/${currentVersion.id}` : '';
 
+        const urlPrefix = this.getUrlPrefix(history.location.pathname);
+
         if (event.preventDefault) {
             event.preventDefault();
         }
 
-        history.push(`${history.location.pathname}/versions${fileVersionSlug}`);
+        history.push(`/${urlPrefix}/versions${fileVersionSlug}`);
```

Comment:
This was previously assuming that the path had no deeplink, so for example `/activity` or `/details`, but now with these deeplinks this was breaking by appending the pathname with `versions`. For example, when clicking from a deeplinked comment the pathname is `/activity/comments/12345` so this was resulting in `/activity/comments/12345/versions/6789`, which is obviously not what we want here.

Reply:
It seems like we're going through a lot of trouble to have the `/tasks` and `/comments` subpaths. Would it be simpler to make it a compound parameter, such as `/activity/task_12345`? React Router supports regexes for parameters.